content/source-git/design.md

```diff
@@ -0,0 +1,189 @@
+known to require further work to be better defined.
+
+**Authors**: Stef Walter, Tomas Tomecek  
+**Updates by:** Hunor Csomortáni
+
+## What is source-git?
+
+Source-git is a repository format and the related processes, tooling and bots,
+that are intended to enable using forks of the upstream projects to maintain,
+update and build packages in a distribution. By "distribution" we mean
+distributions in the RHEL ecosystem (Fedora Linux, CentOS Stream and RHEL), but
+the tools and processes probably could be applied to any RPM based distribution.
+
+## Motivation
+
+Traditionally dist-git is the format used to maintain, develop and release
+software in these distributions. In dist-git the source archive of an upstream
+release is stored in the lookaside cache, while downstream (distribution
+specific) changes are checked-in as patch files in Git. This layout resembles
+SRPMs, and so it's easily consumed by build systems, but it makes it somewhat
+difficult for humans to make sense of the content. Additionally, transforming
+upstream content (which most of the time originates from a Git repo) to dist-git
+has become a tedious activity, with a plethora of tooling available to do very
+similar things.
+
+The dist-git format also has the side effect of making the adoption of modern
+Git-workflows somewhat more difficult (reviewing changes in patch-files requires
+a higher cognitive effort) and raising the bar for new contributors, who know
+how to contribute using Git, but need to learn about dist-git before touching
+any package.
+
+With source-git the goal is to:
+- Enable using these well-known Git-workflows in packaging activities.
+- Automate and standardize the tedious task of converting from one repository
+  format to the other.
+  
+If you think about it, "source-git" is really just good old plain "Git", used as
+it meant to be used by Linus.
+
+## Premises
+
+One of the fundamentally useless manual activities when maintaining a package
+in Fedora is transforming source code from one Git repository format to another.
+Git is distributed. Dist-git content is mostly boilerplate or regurgitated data.
+
+Using an upstream format during packaging makes collaboration easier, and lowers
+the entry barrier for new contributors by enabling a development workflow which
+they are already familiar with.
+
+Linux distributions gain an advantage from having patches incorporated upstream
+and not carrying them downstream.
+
+Human effort should not be focused on repetitive, automatable tasks related to
+churn and moving code around.
+
+Dist-git is used as a store of state for build tools (like Koji). Reinventing
+dist-git itself fundamentally, would mean reinventing a lot of tooling.
+
+## An addon to dist-git
+
+We recognize, that an extensive ecosystem of tooling and services was developed
+to work with dist-git. Because of this, replacing dist-git with source-git is
+not feasible.
```

Comment:
although this is definitely the long-term goal

could we reword to something like

> replacing dist-git with source-git is a gargantuan effort.

Reply:
You mean to express that replacing it is not feasible in the immediate future, but it can be a goal in the long run?

Reply:
Yes, that was the point I tried to make